Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -44,8 +44,23 @@ void UPSSpotComponent::BeginPlay()
 // Clears all transient data created by this component.
 void UPSSpotComponent::OnUnregister()
 {
+	// reset back to initial state. By default, the spot is unlocked 
+	constexpr bool bSpotUnlocked = true;
+	if (PlayerSpotOnLevelInternal)
+	{
+		PlayerSpotOnLevelInternal->SetActive(bSpotUnlocked);
+		PlayerSpotOnLevelInternal = nullptr;
+	}
+
+	if (SaveGameInstanceInternal)
+	{
+		SaveGameInstanceInternal->ConditionalBeginDestroy();
+		SaveGameInstanceInternal = nullptr;
+	}
+
+	UPSWorldSubsystem::Get().RemoveProgressionSystemSpot(this);
```

Comment:
- It's potentially dangerous code, as it can crash if world subsystem is destroyed by that time (that might happen during unload, order is not guaranteed). Instead, it's just enough to unregister ALL cached PS Spots in World Subsystem. Just calling `DestroyStarActors` by world subsystem is already enough and more safe
- Please remove `RemoveProgressionSystemSpot` as unused then

Reply:
removed: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/436cd8ffaacfbaf68caa7be0f2a227a7f4378364